crypto/bio/bss_sock.c

```diff
@@ -64,7 +65,22 @@ BIO *BIO_new_socket(int fd, int close_flag)
     if (ret == NULL)
         return NULL;
     BIO_set_fd(ret, fd, close_flag);
-    return ret;
+#ifndef OPENSSL_NO_KTLS
+    {
+        /*
+         * The new socket is created successfully regardless of ktls.
+         * Thus, we don't propogate this error.
+         * However, future calls to ktls will fail.
+         */
+        int rc = ktls_enable(fd);
```

Comment:
ktls_enable calls the TCP_ULP socket option. This socket option doesn't change the behaviour of the TCP socket, except changing the TCP setsockopt handler to enable the processing of SOL_TLS options. Only the TLS_TX socket option changes the send/sendmsg handlers. We set it for all sockets because it allows for transparent support of all existing applications. Another option is to call enable_ktls when handling BIO_CTRL_SET_OFFLOAD_TX.

Reply:
If we keep it here, then at least a comment to say why it isn't a problem for non-tls sockets might be in order.

Reply:
Done. I'll also add some description to include/internal/ktls.h to describe these functions.